Fixes #20352 - backup deleted tasks

mbacovsky opened the pull request:
This patch uses and extends https://github.com/Dynflow/dynflow/pull/241.
- adds backup of foreman tasks and ensures it lands in the same dir as the backup of related dynflow objects
- adds a way to configure the backups (the default dir is `./backup` for now though. Should it be set during packaging?)

```
cat >> /etc/foreman/plugins/foreman-tasks.yaml <<EOF

  :backup:
    :backup_deleted_plans: true
    :backup_dir: /var/lib/foreman/tasks-backup
```

adamruzicka commented on lib/foreman_tasks/cleaner.rb:
```diff
@@ -90,13 +93,29 @@ def tasks
 
-    def delete_tasks(chunk)
-      ForemanTasks::Task.where(:id => chunk.map(&:id)).delete_all
+    def delete_tasks(chunk, backup_dir = ForemanTasks.dynflow.world.persistence.current_backup_dir)
```
This breaks compatibility with remote execution's extensions for the cleaner, I'm not against the change, just making a note of it

iNecas replied:
It seems to me having the `backup_dir` as instance variable configured in the initializer would help here, wouldn't it? It would also reduce some code duplicity.

mbacovsky replied:
@iNecas makes sense as Cleaner instance has short life

mbacovsky commented:
Moved backup_dir to instance variable

adamruzicka commented on lib/foreman_tasks/cleaner.rb:
```diff
@@ -91,12 +95,28 @@ def tasks
     end
 
-    def delete_dynflow_plans(chunk)
+    def delete_dynflow_plans(chunk, backup_dir = ForemanTasks.dynflow.world.persistence.current_backup_dir)
```
Any reason for passing the `backup_dir` explicitly instead of using `@backup_dir`?

mbacovsky commented:
Updated - removed backup_dir also from delete_dynflow_plans, fixed rubocop.
The tests are failing due to missing related changes in Dynflow. It should wok fine once dynflow PR is merged. Should I uopdate the dependency on dynflow?

adamruzicka commented:
Tested with https://github.com/Dynflow/dynflow/pull/241 and it works as expected. Please bump the dependency on dynflow. Tests passed locally just fine, but we'll wait for Mr. Jenkins to confirm that.

iNecas commented:
@mbacovsky there are conflicts with the current maser, Please resolve

adamruzicka commented on lib/foreman_tasks/dynflow/configuration.rb:
```diff
@@ -6,6 +6,27 @@
+    def world_config
+      super.tap do |config|
+        config.backup_deleted_plans = backup_settings[:backup_deleted_plans]
+        config.backup_dir          = backup_settings[:backup_dir]
```
Nitpick: needs one extra space to align the `=`

adamruzicka commented:
Looks good to me, let's wait what Jenkins has to say

adamruzicka commented:
@iNecas anything to change here or can we get it in?

iNecas commented:
Closing in favor of  #277
